code/game/machinery/constructable_frame.dm

```diff
@@ -215,14 +215,16 @@ to destroy them and players will be able to make replacements.
 	desc = "A blank circuitboard ready for design."
 	icon = 'icons/obj/module.dmi'
 	icon_state = "blank_mod"
-	var/datum/circuits/local_fuses = null
+	//var/datum/circuits/local_fuses = null
+	var/list/allowed_boards = list("autolathe"=/obj/item/weapon/circuitboard/autolathe,"intercom"=/obj/item/weapon/intercom_electronics,"conveyor"=/obj/item/weapon/circuitboard/conveyor,"air alarm"=/obj/item/weapon/circuitboard/air_alarm,"fire alarm"=/obj/item/weapon/circuitboard/fire_alarm,"airlock"=/obj/item/weapon/circuitboard/airlock,"APC"=/obj/item/weapon/circuitboard/power_control,"vendomat"=/obj/item/weapon/circuitboard/vendomat,"microwave"=/obj/item/weapon/circuitboard/microwave)
+	var/list/board_names = list("autolathe","intercom","conveyor","air alarm","fire alarm","airlock","APC","vendomat","microwave")

```

Comment:
You shouldn't need this list.
It should be possible to pick from allowed_boards, due to the way Byond handles lists.